### ISSUE TITLE
[MPLUGIN-271] Usage of defaultValues for a List<..> Parameter seemed not to be working

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-plugin-tools#591]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** opened **[MPLUGIN-271](https://issues.apache.org/jira/browse/MPLUGIN-271?redirect=false)** and commented

Currently the following seemed to be not working:

```java
@Parameter( defaultValue = "*/pom.xml" )
private List<String> pomIncludes;
```

---

**Affects:** 3.3

**Issue Links:**
- [MNG-5440](https://issues.apache.org/jira/browse/MNG-5440) default-value on mojo parameter of type collection or array effectively make parameter read-only

- [MPMD-206](https://issues.apache.org/jira/browse/MPMD-206) Make the sourceDirectories configurable
 (_**"is depended upon by"**_)

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

Another issue on the same note.

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte) Do you know where the processing of a plugin configuration xml-snippet is done? You mentioned plexus, but which part of plexus?

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

A fix has been committed to Sisu Plexus and is scheduled for the 1.0.0 release.

**Comment by jira-importer** [comment]
**[Igor Fedorenko](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=igorf)** commented

This bug is not specific to maven-plugin-annotations, I think it makes sense to move it to MNG jira, where users can find it more easily.

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

[Igor Fedorenko](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=igorf) There is already an issue in MNG, that I've linked to this one.

**Comment by jira-importer** [comment]
**[Stuart McCulloch](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mcculls)** commented

I've cherry-picked the fix back to Sisu/Plexus 0.3.2 which is now available on central.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Not a problem of maven-plugin-plugin, but of Maven core. Fixed in Maven 3.3.9
